### PR TITLE
32 edit delete profil page

### PR DESCRIPTION
**Pull request by Lith3**
Allow the user to modify their information or delete their profile.
Improve the visual part of the profile and booking page (which uses the same components as the profile page)

[US link](https://github.com/WildCodeSchool-2024-02/JS-lyon-0224-P3-cretchum/issues/67)

### CODE REVIEW

**Comment by StoicFrenchDev** [comment]
Have you tried deleting a profile that is linked to a home_structure ? @Lith3

File: client/src/components/profile/editable_field/EditableField.jsx [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
         type="text"
         defaultValue={value}
-        readOnly={isEditMode !== true}
-        onChange={isEditMode === true ? onChange : undefined}
-        className={`${styles.input} ${isEditMode !== true ? styles.readOnlyInput : ""}`}
+        readOnly={isEditMode === false}
+        onChange={
+          isEditMode === true ? (e) => onChange(e, valueName) : undefined
+        }
+        className={`${styles.input} ${!isEditMode ? styles.readOnlyInput : ""}`}
```
Comment (wildmatthieu) [inline client/src/components/profile/editable_field/EditableField.jsx:19]: expliciter la condition : 
`isEditMode !== ___ ? styles...`

File: server/app/controllers/usersActions.js [hunk 2]
```diff
@@ -102,7 +101,11 @@ const checkLog = async (req, res, next) => {
     const user = await tables.user.login(mail);
 
     // Check that the user exists and that the password is correct
-    if (user !== null && user !== undefined && await bcrypt.compare(password, user.password)) {
+    if (
+      user !== null &&
+      user !== undefined &&
+      (await bcrypt.compare(password, user.password))
```
Comment (wildmatthieu) [inline server/app/controllers/usersActions.js:107]: expliciter la condition : 
`await bcrypt.compare(password, user.password) === ???`

**Review by wildmatthieu — CHANGES_REQUESTED** [review]

**Review by HazeFury — APPROVED** [review]
gg

**Pull request merged**